pkg/virtual/framework/wrappers/rbac/merging.go

```diff
@@ -0,0 +1,469 @@
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedRoleBindingLister) RoleBindings(namespace string) rbaclisters.RoleBindingNamespaceLister {
+	aggregatedListers := make([]rbaclisters.RoleBindingNamespaceLister, 0)
+	for _, inf := range l.listers {
+		aggregatedListers = append(aggregatedListers, inf.RoleBindings(namespace))
+	}
+	return &mergedRoleBindingNamespaceLister{
+		listers: aggregatedListers,
+	}
+}
+
+func (l *mergedRoleBindingNamespaceLister) List(selector labels.Selector) (ret []*rbacv1.RoleBinding, err error) {
+	aggregatedList := make([]*rbacv1.RoleBinding, 0)
+	for _, lister := range l.listers {
+		list, err := lister.List(selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedRoleBindingNamespaceLister) Get(name string) (*rbacv1.RoleBinding, error) {
+	var errorHolder error
+	for _, lister := range l.listers {
+		item, err := lister.Get(name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			return item, nil
+		}
+	}
+	return nil, errorHolder
+}
+
+func MergedRoleInformer(informers []rbacinformers.RoleInformer) rbacinformers.RoleInformer {
+	return &mergedRoleInformer{
+		informers: informers,
+	}
+}
+
+var _ rbacinformers.RoleInformer = (*mergedRoleInformer)(nil)
+var _ rbaclisters.RoleLister = (*mergedRoleLister)(nil)
+var _ rbaclisters.RoleNamespaceLister = (*mergedRoleNamespaceLister)(nil)
+
+type mergedRoleInformer struct {
+	informers []rbacinformers.RoleInformer
+}
+
+type mergedRoleLister struct {
+	listers []rbaclisters.RoleLister
+}
+
+type mergedRoleNamespaceLister struct {
+	listers []rbaclisters.RoleNamespaceLister
+}
+
+func (i *mergedRoleInformer) Informer() cache.SharedIndexInformer {
+	panic("not implemented")
+}
+
+func (i *mergedRoleInformer) Lister() rbaclisters.RoleLister {
+	aggregatedListers := make([]rbaclisters.RoleLister, 0)
+	for _, inf := range i.informers {
+		aggregatedListers = append(aggregatedListers, inf.Lister())
+	}
+	return &mergedRoleLister{
+		listers: aggregatedListers,
+	}
+}
+
+func (l *mergedRoleLister) List(selector labels.Selector) (ret []*rbacv1.Role, err error) {
+	aggregatedList := make([]*rbacv1.Role, 0)
+	for _, lister := range l.listers {
+		list, err := lister.List(selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedRoleLister) ListWithContext(ctx context.Context, selector labels.Selector) (ret []*rbacv1.Role, err error) {
+	aggregatedList := make([]*rbacv1.Role, 0)
+	for _, lister := range l.listers {
+		list, err := lister.ListWithContext(ctx, selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
```

Comment:
do we need merging for roles? Are there roles in the bootstrap policy?

Reply:
There are quite a few of them:

```
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: system::leader-locking-kube-controller-manager
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: system::leader-locking-kube-scheduler
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: system:controller:bootstrap-signer
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: extension-apiserver-authentication-reader
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: system:controller:bootstrap-signer
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: system:controller:cloud-provider
Apr  6 10:38:19.373: INFO: Bootstrap policy Role: system:controller:token-cleaner
```

I'm not sure if we're interested in those but I think we could leave the merging mechanism for them as well (at least for now). WDYT?